src/main/java/com/ss/training/utopia/orchestrator/counter/controller/CounterOrchestrator.java

```diff
@@ -0,0 +1,125 @@
+import com.ss.training.utopia.orchestrator.entity.Flight;
+import com.ss.training.utopia.orchestrator.security.User;
+
+/**
+ * @author Justin O'Brien
+ */
+@RestController
+@CrossOrigin
+@RequestMapping("/counter")
+public class CounterOrchestrator {
+
+	@Autowired
+	RestTemplate template;
+
+	private final String baseUrl = "http://localhost:8081/counter";
+
+	@RequestMapping(method = RequestMethod.HEAD, path = "/traveler/{username}")
+	public ResponseEntity<Object> userIsTraveler(@PathVariable String username, RequestEntity<?> request) {
+		try {
+			return template.exchange(baseUrl + "/traveler/" + username, HttpMethod.HEAD, request, Object.class);
+		} catch (RestClientResponseException e) {
+			return new ResponseEntity<Object>(null, HttpStatus.valueOf(e.getRawStatusCode()));
+		}
+	}
+
+	@GetMapping("/users/{username}")
+	public ResponseEntity<User> getUser(@PathVariable String username, RequestEntity<?> request) {
+		try {
+			return template.exchange(baseUrl + "/users/" + username, HttpMethod.GET, request, User.class);
+		} catch (RestClientResponseException e) {
+			return new ResponseEntity<User>((User) null, HttpStatus.valueOf(e.getRawStatusCode()));
+		}
+	}
+
+	@GetMapping("/airports")
+	public ResponseEntity<Airport[]> getAllAirports(RequestEntity<?> request) {
+		try {
+			return template.exchange(baseUrl + "/airports", HttpMethod.GET, request, Airport[].class);
+		} catch (RestClientResponseException e) {
+			return new ResponseEntity<Airport[]>((Airport[]) null, HttpStatus.valueOf(e.getRawStatusCode()));
+		}
+	}
+
+	@GetMapping("/flights/cancellable/traveler/{travelerId}")
+	public ResponseEntity<Flight[]> getCancellablyBookedFlights(@PathVariable Long travelerId,
+			RequestEntity<?> request) {
+		try {
+			return template.exchange(baseUrl + "/flights/cancellable/traveler/" + travelerId, HttpMethod.GET, request,
+					Flight[].class);
+		} catch (RestClientResponseException e) {
+			return new ResponseEntity<Flight[]>((Flight[]) null, HttpStatus.valueOf(e.getRawStatusCode()));
+		}
+	}
+
+	@PutMapping("/bookings/traveler/{travelerId}/flight/{flightId}")
+	public ResponseEntity<Object> cancelBooking(@PathVariable Long travelerId, @PathVariable Long flightId,
+			RequestEntity<?> request) {
+		try {
+			return template.exchange(baseUrl + "/bookings/traveler/" + travelerId + "/flight/" + flightId,
+					HttpMethod.PUT, request, Object.class);
+		} catch (RestClientResponseException e) {
+			return new ResponseEntity<Object>(null, HttpStatus.valueOf(e.getRawStatusCode()));
+		}
+	}
+
+	@PostMapping("/user")
+	public ResponseEntity<User> createUser(RequestEntity<?> request) {
+		try {
+			return template.exchange(baseUrl + "/user", HttpMethod.POST, request, User.class);
+		} catch (RestClientResponseException e) {
+			return new ResponseEntity<User>((User) null, HttpStatus.valueOf(e.getRawStatusCode()));
+		}
+	}
+
+	@RequestMapping(method = RequestMethod.HEAD, path = "/user/{username}")
```

Comment:
Did you get the HEAD to work with the front-end?

Reply:
Nope, still getting the same error. I just Slacked the CORS configuration to Chris.